### ISSUE TITLE
cdklocal deploy fails with error

### ISSUE THREAD

**Issue by hackerhasid** [localstack/localstack#3837]
# Type of request: This is a ...

[x ] bug report
[ ] feature request

# Detailed description

Deploy command returns failure status and error. However most (maybe all?) resources are created successfully (though names seem off, see comment below)

I am running MacOS Big Sur, docker 20.10.0, localstack 0.12.9.1, cdk 1.97.0, latest cdklocal from npm (installed today).

First I ran

> cdklocal bootstrap

which ran just fine, with output:

```
 ⏳  Bootstrapping environment aws://000000000000/us-east-1...
CDKToolkit: creating CloudFormation changeset...
5:30:45 PM | UPDATE_IN_PROGRESS   | AWS::CloudFormation::Stack | UsePublicAccessBlockConf
iguration
 ✅  Environment aws://000000000000/us-east-1 bootstrapped.
```

Then I attempted to run:

> cdklocal deploy

which had quite a lot of output and seemingly created the stack resources but fails at the end with

```
5:50:17 PM | CREATE_FAILED        | AWS::CloudFormation::Stack  | MyAppStack


 ❌  InjurycomAppStack failed: Error: The stack named MyAppStack failed to deploy: CREATE_FAILED
    at Object.waitForStackDeploy (/usr/local/lib/node_modules/aws-cdk/lib/api/util/cloudformation.ts:307:11)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at Object.deployStack (/usr/local/lib/node_modules/aws-cdk/lib/api/deploy-stack.ts:297:26)
    at CdkToolkit.deploy (/usr/local/lib/node_modules/aws-cdk/lib/cdk-toolkit.ts:180:24)
    at initCommandLine (/usr/local/lib/node_modules/aws-cdk/bin/cdk.ts:208:9)
The stack named MyAppStack failed to deploy: CREATE_FAILED
```

The full output is [here](https://github.com/localstack/localstack/files/6267753/cdklocal.deploy.txt)

I also captured the debug log from localstack [here](https://github.com/localstack/localstack/files/6267747/localstack.debug.txt)

Note that after this fails if I run

> awslocal dynamodb list-tables

it shows that the table was created

> awslocal apigateway get-rest-apis

shows that the api was created

> awslocal lambda list-functions

shows that the lambda functions were created

## Expected behavior

Stack deploys successfully and exits with 0 status code

## Actual behavior

Stack fails to deploy and exits with non-0 status code

# Steps to reproduce

I used this simple (Typescript) stack to demonstrate the issue: https://gist.github.com/hackerhasid/87e26fe810c179223ad61a3afe019449

I ran

> cdklocal bootstrap

then

> cdklocal deploy

## Command used to start LocalStack

> DEBUG=1 localstack start 

## Client code (AWS SDK code snippet, or sequence of "awslocal" commands)

n/a

**Comment by hackerhasid** [comment]
Found another oddity that may be related. The DynamoDB table was created but the name seems off.

Notice that I set an environment variable here `TABLE_FORM_DEFINITIONS ` that's passed to lambda expressions: https://gist.github.com/hackerhasid/87e26fe810c179223ad61a3afe019449#file-myappstack-ts-L20 . The table name there is `FormDefinitions149A813C` and that's what in the environment variables. But running
> awslocal dynamodb list-tables

outputs
```{
    "TableNames": [
        "MyAppStack-FormDefinitions149A813C-c66dcf8d"
    ]
}```

which means that my lambda expressions don't have the right table names to work with. so even though the table is there it's not there as far as my app is concerned.
